--- a/src/app/toys.c
+++ b/src/app/toys.c
@@ -172,7 +172,7 @@ STATIC_FUNC s32 toyProxEventCb(s32 proxId, s32 eventType, s32 clientData) {
     switch (eventType) {
     case 0:
         prox = proxAnimGetHandle(proxId);
-        if (toy_80347C2C(prox->pos, (Toy*)clientData) == -1) {
+        if (toy_80347C2C(prox->pos, (Toy*)PW64_PTR(clientData)) == -1) {
             return -1;
         }
         break;
@@ -187,7 +187,7 @@ STATIC_FUNC s32 toyProxAnimCb(s32 proxId, UNUSED f32 timeout, s32 clientData) {
     Toy* toy;
     s32 res;
 
-    toy = (Toy*)clientData;
+    toy = (Toy*)PW64_PTR(clientData);
     res = 0;
     if ((proxAnimGetRange(proxId) > 100.0f) && (toy->toyType != TOY_MOUNT_RUSHMORE)) {
         res = 2;
@@ -206,7 +206,7 @@ void toyLoad(LevelTOYS* lvlToy) {
     }
 
     toy = &sToys[sToyCount];
-    sToys[sToyCount].unk0 = proxAnimAddCallback(toyProxAnimCb, toyProxEventCb, lvlToy->pos, 100.0f, 0.0f, (s32)toy);
+    sToys[sToyCount].unk0 = proxAnimAddCallback(toyProxAnimCb, toyProxEventCb, lvlToy->pos, 100.0f, 0.0f, (s32)PW64_U32(toy));
     if (sToys[sToyCount].unk250 == 1.0f) {
         uvSobjModel(toy->terraId, MODEL_STONE_WARIO_HEAD);
     }
