--- a/src/libultra/audio/env.c
+++ b/src/libultra/audio/env.c
@@ -153,7 +153,7 @@ Acmd *alEnvmixerPull(void *filter, s16 *outp, s32 outCount, s32 sampleOffset,
                       } data;                    
                       data.f = param->pitch;
                       (*f->source->setParam)(f->source, AL_FILTER_SET_PITCH,
-                                            (void *)data.i);
+                                            (void *)(__INTPTR_TYPE__)data.i); /* PW64: int packed into void* */
                   }
                   
               }
@@ -277,3 +277,4 @@ Acmd *alEnvmixerPull(void *filter, s16 *outp, s32 outCount, s32 sampleOffset,
               (*e->filter.setParam)(&e->filter, e->ctrlList->type,
-                                    (void *) e->ctrlList->data.i);
+                                    (void *)(__INTPTR_TYPE__) e->ctrlList->data.i);
               break;
         }
         loutp  += (samples<<1);
