--- a/src/libultra/audio/load.c
+++ b/src/libultra/audio/load.c
@@ -100,7 +100,7 @@ Acmd *alAdpcmPull(void *filter, s16 *outp, s32 outCount, s32 sampleOffset, Acmd
          * Now fix up state info to reflect the loop start point
          */
         f->lastsam = f->loop.start &0xf;
-        f->memin = (s32) f->table->base + ADPCMFBYTES *
+        f->memin = (s32) PW64_U32(f->table->base) + ADPCMFBYTES *
             ((s32) (f->loop.start>>LFSAMPLES) + 1);
         f->sample = f->loop.start;
 
@@ -162,7 +162,7 @@ Acmd *alAdpcmPull(void *filter, s16 *outp, s32 outCount, s32 sampleOffset, Acmd
      * overFlow is the number of bytes past the end
      * of the bitstream I try to generate
      */
-    overFlow = f->memin + nbytes - ((s32) f->table->base + f->table->len);
+    overFlow = f->memin + nbytes - ((s32) PW64_U32(f->table->base) + f->table->len);
     if (overFlow < 0)
         overFlow = 0;
     nOver = (overFlow/ADPCMFBYTES)<<LFSAMPLES;
@@ -247,7 +247,7 @@ Acmd *alRaw16Pull(void *filter, s16 *outp, s32 outCount, s32 sampleOffset, Acmd
          */
         *outp += dramAlign;
         
-        f->memin = (s32) f->table->base + (f->loop.start<<1);
+        f->memin = (s32) PW64_U32(f->table->base) + (f->loop.start<<1);
         f->sample = f->loop.start;
         op = *outp;
         
@@ -308,7 +308,7 @@ Acmd *alRaw16Pull(void *filter, s16 *outp, s32 outCount, s32 sampleOffset, Acmd
      */
 
     nbytes = outCount<<1;
-    overFlow = f->memin + nbytes - ((s32) f->table->base + f->table->len);
+    overFlow = f->memin + nbytes - ((s32) PW64_U32(f->table->base) + f->table->len);
     if (overFlow < 0)
         overFlow = 0;
     if (overFlow > nbytes)
@@ -359,7 +359,7 @@ alLoadParam(void *filter, s32 paramID, void *param)
     switch (paramID) {
         case (AL_FILTER_SET_WAVETABLE):
             a->table = (ALWaveTable *) param;
-            a->memin = (s32) a->table->base;
+            a->memin = (s32) PW64_U32(a->table->base);
             a->sample = 0;
             switch (a->table->type){
                 case (AL_ADPCM_WAVE):
@@ -415,7 +415,7 @@ alLoadParam(void *filter, s32 paramID, void *param)
 	    /* Get loop info according to table type. */
 	    if (a->table)
 	    {
-		a->memin  = (s32) a->table->base;
+		a->memin  = (s32) PW64_U32(a->table->base);
 		if (a->table->type == AL_ADPCM_WAVE)
 		{
 		    if (a->table->waveInfo.adpcmWave.loop)
